ewms_pilot/pilot.py

```diff
@@ -128,57 +164,100 @@ def read_from_subproc(
     Also, dump to a file for debugging (if not "").
     """
     if not fpath_from_subproc.exists():
-        LOGGER.error("Out file was not written for in-payload")
-        raise RuntimeError("Out file was not written for in-payload")
+        msg = "Out-file cannot be found"
+        LOGGER.error(msg)
+        raise FileNotFoundError(msg)
 
     out_msg = file_reader(fpath_from_subproc)
 
-    # persist the file?
-    if debug_subdir:
-        # fpath_from_subproc.rename(debug_subdir / fpath_from_subproc.name)  # mv
-        # NOTE: https://github.com/python/cpython/pull/30650
-        shutil.move(fpath_from_subproc, debug_subdir / fpath_from_subproc.name)
-    else:
-        fpath_from_subproc.unlink()  # rm
+    mv_or_rm_file(fpath_from_subproc, debug_subdir)
 
     return out_msg
 
 
-def process_msg(
+async def _stream_to_file(stream: Optional[asyncio.StreamReader], bin_out: Any) -> None:
+    if not stream:
+        return
+    while not stream.at_eof():
+        if data := await stream.readline():
+            print(data, file=bin_out)
+
+
+async def process_msg_task(
     in_msg: Any,
     cmd: str,
     subproc_timeout: Optional[int],
-    fpath_to_subproc: Path,
-    fpath_from_subproc: Path,
+    #
+    ftype_to_subproc: FileType,
+    ftype_from_subproc: FileType,
+    #
     file_writer: Callable[[Any, Path], None],
     file_reader: Callable[[Path], Any],
+    #
     debug_dir: Optional[Path],
+    pub: mq.queue.QueuePubResource,
 ) -> Any:
-    """Process the message in a subprocess using `cmd`."""
+    """Process the message's task in a subprocess using `cmd` & respond."""
+    task_id = str(time.time())
```

Comment:
Note that this relies on the fact that `time()` is microsecond precision, and that computers are "slow" enough not to have two tasks launch in the same microsecond.

A better design might be to use a uuid.

Reply:
how about c71cf1b0f5bce805d33e95ee91fbed4757ab514b?

Reply:
Works for me.

Reply:
the message ids are messy for some of the brokers, so I went with using a uuid